snowflakedb/snowflake-cli
# Support generate project definition docs

### Pre-review checklist
   * N/A I've added or updated integration tests to verify correctness of my new code.
   * [x] I've confirmed that my changes are working by executing CLI's commands manually on MacOS.
   * [x] I've confirmed that my changes are working by executing CLI's commands manually on Windows.
   * N/A I've described my changes in the release notes.

### Changes description

- Add a feature to generate descriptions from [ProjectDefinition](https://github.com/snowflakedb/snowflake-cli/blob/5e98392cfcac58cb02436950769ce0afe5ba01b8/src/snowflake/cli/api/project/schemas/project_definition.py#L74), This will create separate files for each project type, resulting in a total of 4 new files being generated.
- Modify the generate method to split documents by commands and project definition.
<img width="383" alt="image" src="https://github.com/snowflakedb/snowflake-cli/assets/127370564/5cab3eb4-d4da-4f2e-90df-c28f1f4335b0">

### Tests
- Add a test to verify the output using a mocked Project Definition.
- Add test to ensure the generated json correspond to the model information.
- Add test to ensure there are a generated file for each command. 

### Example output
[definition_native_app.txt](https://github.com/user-attachments/files/15972308/definition_native_app.txt)
